qiskit/extensions/unitary.py

```diff
@@ -160,10 +160,14 @@ def control(self, num_ctrl_qubits=1, label=None, ctrl_state=None):
         mat = self.to_matrix()
         cmat = _compute_control_matrix(mat, num_ctrl_qubits, ctrl_state=None)
         try:
-            cmat_def = qs_decomposition(cmat)
-        except QSDError:
+            cmat_def = qs_decomposition(cmat, opt_a1=True, opt_a2=True)
+        except QSDError as err:
+            import warnings
             from qiskit.extensions.quantum_initializer.isometry import Isometry
 
+            warnings.warn(
+                f"Error in qs_decomposition. Trying isometry decomposition.\n{err}", UserWarning
```

Comment:
Is `UserWarning` the right exception to raise here, given that this isn't generated from user code? Maybe `RuntimeWarning`?

ref: https://docs.python.org/3/library/exceptions.html#UserWarning

Reply:
Also, is this user actionable? i.e. does it represent something wrong in the user's code, or a bug in Qiskit? If not, maybe logging this would be a better option.